Agda.cabal

```diff
@@ -608,57 +608,90 @@ library
   other-modules:    Paths_Agda
 
   -- NOTE: If adding flags, also add to `.ghci-8.0`
-  ghc-options: -fexpose-all-unfoldings
-               -fspecialise-aggressively
-               -fprint-potential-instances
-               -- Initially, we disable all the warnings.
-               -w
-               -Werror
-               -- This option must be the first one after disabling the
-               -- warnings. See Issue #2094.
-               -Wunrecognised-warning-flags
-               -Wdeprecated-flags
-               -Wderiving-typeable
-               -Wdodgy-exports
-               -Wdodgy-foreign-imports
-               -Wdodgy-imports
-               -Wduplicate-exports
-               -Wempty-enumerations
-               -Widentities
-               -Wincomplete-patterns
-               -Winline-rule-shadowing
-               -Wmissing-fields
-               -Wmissing-methods
-               -Wmissing-pattern-synonym-signatures
-               -Wmissing-signatures
-               -Wnoncanonical-monad-instances
-               -Wnoncanonical-monoid-instances
-               -Woverflowed-literals
-               -Woverlapping-patterns
-               -Wsemigroup
-               -Wtabs
-               -Wtyped-holes
-               -Wunrecognised-pragmas
-               -Wunticked-promoted-constructors
-               -Wunused-do-bind
-               -Wunused-foralls
-               -Wwarnings-deprecations
-               -Wwrong-do-bind
+  if impl(ghc <= 8.0.2)
+    ghc-options: -fexpose-all-unfoldings
+                 -fspecialise-aggressively
+                 -fprint-potential-instances
+                 -- Initially, we disable all the warnings.
+                 -w
+                 -Werror
+                 -- This option must be the first one after disabling the
+                 -- warnings. See Issue #2094.
+                 -Wunrecognised-warning-flags
```

Comment:
If I understand your use-case correctly, you want to be able to add custom warnings via your local config, but without causing a compiler error.

Instead of all this duplication, perhaps appending `[-Wwarn`](https://downloads.haskell.org/~ghc/8.4.2/docs/html/users_guide/using-warnings.html#ghc-flag--Wwarn) as the last flag would do the trick, so that the additional flags added by your local config would not be treated as errors.

Reply:
The maintainers appear to want the compilation/build to fail upon certain warnings only. The way the original `Agda.cabal` was constructed, it conflicted with the global `~/.cabal/config` setting of `-Wall`. 

I am not familiar with `-Wwarn`, and with how it relates to `-Wall` and `-Werror`.

Plus, if the maintainers want to fail only upon some warnings, but not upon the others - it makes sense to explicitly say exactly that - "fail upon this set of  warnings", which is what this PR is doing.

I would simply change
```
  -Werror
  -Wunrecognized-warning-flags
. . . . .
```
to
```
  -Werror=unrecognized-warning-flags
  . . . . .
```

Apparently, GHC-8.0.2 does not support that form. So, I had to preserve the original way for 8.0.2 and earlier, and adding explicit lists for the later, more capable compilers.

Reply:
Please reread my suggestion. I said _append_, not prepend. The link specifically says of `-Wwarn` _"can be useful to negate a `-Werror` flag"_:
```
-Werror
-Wunrecognized-warning-flags
…
-Wwarn
```

(The docs on that flag are identical for 8.0.2, I just had copied the wrong tab).

I haven't verified that that works, but the docs seem to indicate that it would.

Reply:
You did not make it clear initially *where* `-Wwarn` should be appended. 

Yes, it makes sense in theory. However, `-w` in the `Agda.cabal` did not negate the prior `-Wall` (from my local config). So, I'm not 100% convinced this would work in practice.

Also, I expect 8.0.2 dropped at some point in time (hopefully soon). At this time, 8.0.2-specifc stuff would simply be purged from `Agda.cabal`, and only the clear `-Werror=unrecognized-warning-flags` form would remain.